### PR TITLE
feat(testing): add wdi5 e2e scope

### PR DESCRIPTION
**Pull request by vobu**
adds `wdi5` for e2e tests:
- in TS flavor
- with minimal scope (config + test file)

### CODE REVIEW

**Comment by vobu** [comment]
move to "draft" status, b/c should only be merge once `wdi5` 2.0 is out

**Comment by akudev** [comment]
This "helloworld" repository started out with a different focus and indeed I have had also difficulties to convey this towards @petermuessig  before. I think it works best like this:

Initially this repo was meant to describe the _TypeScript setup as such_. Which additional files are needed, what do they need to contain, to have TS code translated to JS in a UI5 app. Yes, there was also a minimal UI5 app contained, but only to have some code to compile, not to show how TS app content should look. Consequently, the documentation focused on a few steps that created those setup-relevant files like the tsconfig and the babelrc.

Now with the testing content added by Peter and more so with the wdi5 content, it turns more and more into a sample how all kinds of aspects should look inside a UI5 app written in TypeScript, the focus on how the setup works under the hood is gone.

And that's ok. We need such a sample, after all, plus it's getting less and less important to understand how the build pipeline works, with ui5-tooling-transpile wrapping the transpilation and making configuration unnecessary and also with Easy UI5 setting everything up for most users.

The only thing is that the once central part of this repo - https://github.com/SAP-samples/ui5-typescript-helloworld/blob/main/step-by-step.md - gets more and more anachronistic with this transition. This page was meant to get to successful TS transpilation from scratch, not anything beyond that. It still says "_Note that the scope of this tutorial is the TypeScript setup of a project, not the application code itself._". So we'll need to see how to describe that part and whether this is the place where to also describe how all the parts of the app, incl. testing should be described.

**TL;DR: yep, all is good, go ahead, but I'll need to check where and how to describe best how the transpilation process works under the hood.**

**Comment by vobu** [comment]
> TL;DR: yep, all is good, go ahead

here we are :)

File: step-by-step.md [hunk 1]
```diff
@@ -127,18 +127,16 @@ So we need to add a `tsconfig.json` configuration file to configure the right la
...
->        ],
->        "types": [
->            "@sapui5/ts-types-esm"
+>            "./node_modules/@types",
+>            "./node_modules/@sapui5/types"
 >        ],
 >```
 >
-> Why? TypeScript automatically finds all type definition files in a dependency starting with `@types/...` (i.e. all `.d.ts` files in `node_modules/@types/...`). The jQuery d.ts files are there and found, but the SAPUI5 types are only available in a package starting with `@sapui5/...`, hence TypeScript must be explicitly pointed to these types. As this disables the automatic loading of other types from `node_modules/@types/...`, this path must be given as a type root.
+> Why? TypeScript automatically finds all type definition files in a dependency starting with `@types/...` (i.e. all `.d.ts` files in `node_modules/@types/...`). The jQuery d.ts files are there and found, but the SAPUI5 types are only available in a package starting with `@sapui5/...`, hence TypeScript must be explicitly pointed to these types. As this disables the automatic loading of other types from `node_modules/@types/...`, this path must also be given as a type root.
```
Comment (akudev) [inline step-by-step.md:139]: Hm, after removing the "types" section, this sentence does no longer apply. However, I think the current recommendation is using `types`, not `typeRoots`. Peter?

File: webapp/test/e2e/dialog.test.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+import AppPage from "./pages/AppPage"
```
Comment (akudev) [inline webapp/test/e2e/dialog.test.ts]: The existing files in this repo DO use semicolon line endings. For consistent style I'd say they need to be added.

File: webapp/test/e2e/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "compilerOptions": {
+    "moduleResolution": "node",
+    "types": ["node", "@openui5/types", "@wdio/globals/types", "@wdio/mocha-framework", "wdio-ui5-service/esm"],
```
Comment (akudev) [inline webapp/test/e2e/tsconfig.json]: @openui5/types are not a dependency (@types/openui5 are used)

**Comment by akudev** [comment]
Apart from few minor comments, the code itself is fine for me.

File: package.json [hunk 4]
```diff
@@ -15,26 +15,32 @@
...
+    "eslint": "^8.49.0",
     "karma": "^6.4.2",
     "karma-chrome-launcher": "^3.2.0",
     "karma-coverage": "^2.2.1",
     "karma-ui5": "^3.0.3",
-    "karma-ui5-transpile": "^3.0.1",
+    "karma-ui5-transpile": "^3.2.3",
     "rimraf": "^5.0.1",
-    "typescript": "^5.1.6",
+    "ts-node": "^10.9.1",
```
Comment (akudev) [inline package.json:42]: Care to elaborate on this change? "tsc" is used above, will it still work with ts-node? It's true that typescript is not used otherwise, but then again ts-node also does not appear to be used directly.

Comment (vobu) [reply package.json:42]: `wdi5`/`wdio` uses `ts-node` for on-the-fly transpiling TS-files to JS - including the tests themselves.
dropping `typescript` wasn't intended. 
will double-check together with the few minor corrections mentioned above.

File: webapp/test/e2e/tsconfig.json (hunk 3 above)
Comment (vobu) [reply webapp/test/e2e/tsconfig.json]: fixed

File: webapp/test/e2e/dialog.test.ts (hunk 2 above)
Comment (vobu) [reply webapp/test/e2e/dialog.test.ts]: done - everywhere   
(insert silent weep here)
